rust/src/gildedrose.rs

```diff
@@ -88,12 +86,86 @@ impl GildedRose {
+                    Item::new("Aged Brie", -1, 4),
+                    Item::new("Elixir of the Mongoose", 2, 4),
+                    Item::new("Sulfuras, Hand of Ragnaros", 0, 80),
+                    Item::new("Sulfuras, Hand of Ragnaros", -1, 80),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", 12, 23),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", 7, 50),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", 2, 50),
+                    Item::new("Conjured Mana Cake", 0, 0),
+                ],
+            },
+            Test {
+                days: 10,
+                output: vec![
+                    Item::new("+5 Dexterity Vest", 0, 10),
+                    Item::new("Aged Brie", -8, 18),
+                    Item::new("Elixir of the Mongoose", -5, 0),
+                    Item::new("Sulfuras, Hand of Ragnaros", 0, 80),
+                    Item::new("Sulfuras, Hand of Ragnaros", -1, 80),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", 5, 35),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", 0, 50),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", -5, 0),
+                    Item::new("Conjured Mana Cake", -7, 0),
+                ],
+            },
+            Test {
+                days: 29,
+                output: vec![
+                    Item::new("+5 Dexterity Vest", -19, 0),
+                    Item::new("Aged Brie", -27, 50),
+                    Item::new("Elixir of the Mongoose", -24, 0),
+                    Item::new("Sulfuras, Hand of Ragnaros", 0, 80),
+                    Item::new("Sulfuras, Hand of Ragnaros", -1, 80),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", -14, 0),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", -19, 0),
+                    Item::new("Backstage passes to a TAFKAL80ETC concert", -24, 0),
+                    Item::new("Conjured Mana Cake", -26, 0),
+                ],
+            },
+        ];
 
-        assert_eq!("fixme", rose.items[0].name);
+        for test in tests {
+            let input = vec![
+                Item::new("+5 Dexterity Vest", 10, 20),
+                Item::new("Aged Brie", 2, 0),
+                Item::new("Elixir of the Mongoose", 5, 7),
+                Item::new("Sulfuras, Hand of Ragnaros", 0, 80),
+                Item::new("Sulfuras, Hand of Ragnaros", -1, 80),
+                Item::new("Backstage passes to a TAFKAL80ETC concert", 15, 20),
+                Item::new("Backstage passes to a TAFKAL80ETC concert", 10, 49),
+                Item::new("Backstage passes to a TAFKAL80ETC concert", 5, 49),
+                Item::new("Conjured Mana Cake", 3, 6),
+            ];
+
+            let mut rose = GildedRose::new(input);
+            days_loop(&mut rose, test.days);
+            for i in 0..test.output.len() {
```

Comment:
I believe you can create an iterator from test as its a vector instead of using a for loop so you dont have to worry about getting the length correct
`test.iter()`

Reply:
I think `iter()` only provides the object, and doesn't support an index, so if he needed the index for the comparison object I think it's not an option. Unless there's a way to have it iterate over both data structures or something?

Reply:
It works using iter().enumerate() to get also the index:

` for (i, item) in test.output.iter().enumerate(){`